File: arraysandstrings/URLify.java
package arraysandstrings;

import java.util.Arrays;
import java.util.stream.Collectors;

public class URLify {
    public static void main(String[] args) {
        assert urlify("Mr John Smith     ").equals("Mr%20John%20Smith");
    }

    private static String urlify(String str) {
        String[] arr = str.split(" ");
        String collect = Arrays.stream(arr).filter(s -> !s.isEmpty()).collect(Collectors.joining("%20"));
        System.out.println(collect);
        return collect;
    }
}

File: src/arraysandstrings/URLify.kt
package arraysandstrings

import kotlin.test.assertEquals

fun main() {
    assertEquals("Mr%20John%20Smith", urlify("Mr John Smith     "))
    assertEquals(expected = "", actual = urlify("   "))
    assertEquals("Prashant%20Barahi", urlify("Prashant          Barahi      "))
}

private fun urlify(str: String): String {
    return str.split(" ").filter { it.isNotBlank() }.joinToString("%20")
}
